microsoft/testsuites/dpdk/dpdksuite.py

```diff
@@ -363,6 +375,66 @@ def __init__(
         self.testpmd = _testpmd
         self.node = _node
         self.nic_controller = _node.features[NetworkInterface]
+        self.dmesg = _node.tools[Dmesg]
+        self._last_dmesg = ""
+        test_nic = self.node_nic_info.get_test_nic()[1]
+        # generate hotplug pattern for this specific nic
+        self.vf_hotplug_regex = re.compile(
+            f"{test_nic.upper}: Data path switched to VF: {test_nic.lower}"
+        )
+        self.vf_slot_removal_regex = re.compile(f"VF unregistering: {test_nic.lower}")
+
+    def wait_for_dmesg_output(self, wait_for: str, timeout: int) -> bool:
+        search_pattern = None
+        if wait_for == "AN_DISABLE":
+            search_pattern = self.vf_slot_removal_regex
+        elif wait_for == "AN_REENABLE":
+            search_pattern = self.vf_hotplug_regex
+        else:
+            raise LisaException(
+                "Unknown search pattern specified in "
+                "DpdkTestResources:wait_for_dmesg_output"
+            )
+
+        self.node.log.info(search_pattern.pattern)
+        time_accumulator = 0.0
+        while time_accumulator < timeout:
```

Comment:
The `process.py` has a simpler pattern to check it with lisa's timer.

Reply:
Do you mean perf_timer?

Reply:
Yes, it can count the accumulated value too.

Reply:
perf_timer.Timer() looks good :)